app/controllers/OrganizationController.scala

```diff
@@ -32,20 +39,37 @@ class OrganizationController @Inject()(organizationDAO: OrganizationDAO,
     }
   }
 
-  def getDefault = Action.async { implicit request =>
+  def list: Action[AnyContent] = sil.SecuredAction.async { implicit request =>
+    for {
+      organizations <- organizationDAO.findAll ?~> "organization.list.failed"
+      js <- Fox.serialCombined(organizations)(o => organizationService.publicWrites(o))
+    } yield Ok(Json.toJson(js))
+  }
+
+  def getDefault: Action[AnyContent] = Action.async { implicit request =>
     for {
       allOrgs <- organizationDAO.findAll(GlobalAccessContext) ?~> "organization.list.failed"
       org <- allOrgs.headOption.toFox ?~> "organization.list.failed"
       js <- organizationService.publicWrites(org)(GlobalAccessContext)
     } yield {
-      if (allOrgs.length > 1) // Cannot list organizations if there are multiple ones due to privacy reasons
+      if (allOrgs.length > 1) // Cannot list organizations publicly if there are multiple ones, due to privacy reasons
```

Comment:
IntelliJ wants me to replace the `.length > 1` call with `allOrgs.headOption.isDefined && allOrgs.tail.nonEmpty`. I don't know the real runtime impact.

Reply:
Hmm, I think this sacrifices readability for performance.
In case of very long lists, this definitely makes sense (do not count millions of links just to compare the result with 1), but we expect allOrgs to stay fairly small. So I’d go for the more readable version (also not chosing `lengthCompare`). Also, if there is a sharp increase in org counts, the place to performance-optimize first is above, by not getting all of them from the db.